micro: port operator FILL from lite

rkuester opened the pull request:
For micro, implement the FILL kernel and test, and add both to the build.

This PR contains several commits for readable history and ease of review.

This is PR5 in the sequence of PRs, outlined in issue #45306, to port operator FILL from lite to micro.

rkuester commented:
This PR is stacked atop PR, #45646. It should be rebased and moved from draft to ready status as soon as that PR is merged.

The appropriate sequence of PRs and mapping of commits to PRs in this port of operator FILL was discussed in #45612.

petewarden commented on tensorflow/lite/micro/kernels/fill.cc:
```diff
@@ -0,0 +1,136 @@
+      FillImpl<float>(value, output);
+      break;
+    default:
+      context->ReportError(
```
This will need to be replaced with the macro equivalent.

gbaned commented:
@rkuester This PR is in draft, can you please take a look on it. Thanks!

gbaned commented:
@rkuester  This PR is in draft, can you please take a look on it. Thanks!

rkuester commented:
FWIW, this PR is stacked atop PR #45646 and is awaiting approval thereof before I rebase it and take it out of draft.

rkuester committed:
micro: port operator FILL from lite

Implement appropriately for micro the FILL kernel and test, and
add both to the build.

rkuester committed:
micro: FILL test: fix lifetime of registration object

Fix a bug caught by ASan.

Don't pass a reference to a temporary TfLiteRegistration object
to the runner. Save the registration object.

rkuester committed:
micro: fix Arduino build for FILL

Fix the Arduino build for fill by adding the header in Lite with
the reference implementation to MICROLITE_CC_HDRS in the micro
Makefile.

gbaned commented:
@rkuester Can you please resolve conflicts? Thanks!

rkuester committed:
Merge master into rkuester:feature-micro-add-op-fill-pr5

Resolve conflict in build configuration where other files were
added alphabeticaly close to those for op FILL.
